File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/listener/StringeeCallListener.java
package com.stringee.apptoappcallsample.stringee.listener;

import com.stringee.apptoappcallsample.stringee.common.CallStatus;
import com.stringee.apptoappcallsample.stringee.common.ConnectionState;
import com.stringee.exception.StringeeError;

/** Host-facing callbacks emitted by {@code StringeeCallManager}. */
public interface StringeeCallListener {
    /** Called when the Stringee connection state or connected user changes. */
    default void onConnectionStateChanged(ConnectionState state, String userId) {
    }

    /** Called when the active call state changes. */
    default void onCallStateChanged(CallStatus state) {
    }

    /** Called when a connection, call, push, or notification operation fails. */
    default void onError(String action, StringeeError error) {
    }

    /** Called when Stringee requests a refreshed access token. */
    default void onRequestNewToken() {
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/listener/StringeeCallListener.kt
package com.stringee.kotlin_onetoonecallsample.stringee.listener

import com.stringee.exception.StringeeError
import com.stringee.kotlin_onetoonecallsample.stringee.common.CallStatus
import com.stringee.kotlin_onetoonecallsample.stringee.common.ConnectionState

/** Host-facing callbacks emitted by the public Stringee call facade. */
interface StringeeCallListener {
    /** Called when the Stringee connection state or connected user changes. */
    fun onConnectionStateChanged(state: ConnectionState, userId: String) {
    }

    /** Called when the active call state changes. */
    fun onCallStateChanged(state: CallStatus) {
    }

    /** Called when a connection, call, push, or notification operation fails. */
    fun onError(action: String, error: StringeeError) {
    }

    /** Called when Stringee requests a refreshed access token. */
    fun onRequestNewToken() {
    }
}
